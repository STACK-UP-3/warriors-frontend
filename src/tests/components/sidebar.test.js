import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import MobileSidebar from '../../components/common/mobileSidebar';
import SideMenu from '../../components/common/sideMenu';
import SideInit from '../../components/common/SideInitializer';

describe('Test <Sidebar/> Component', () => {
  it('Should render Mobile Sidebar correctly', () => {
    const mobSide = shallow(<MobileSidebar />);
    expect(toJson(mobSide)).toMatchSnapshot();
  });
  it('Should render Sidemenu correctly', () => {
    const sideMenu = shallow(<SideMenu />);
    expect(toJson(sideMenu)).toMatchSnapshot();
  });
  it('Should render Side Initializer correctly', () => {
    const sideInit = shallow(<SideInit />);
    expect(toJson(sideInit)).toMatchSnapshot();
  });
});
