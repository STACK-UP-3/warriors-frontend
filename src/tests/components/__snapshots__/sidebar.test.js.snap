// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Sidebar/> Component Should render Mobile Sidebar correctly 1`] = `
<div>
  <div
    className="sidenav"
    id="slide-out"
  >
    <_default />
  </div>
</div>
`;

exports[`Test <Sidebar/> Component Should render Side Initializer correctly 1`] = `""`;

exports[`Test <Sidebar/> Component Should render Sidemenu correctly 1`] = `
<div
  className="sideMenuContainer"
>
  <div
    className="profile"
  >
    <img
      src={
        Object {
          "process": [Function],
        }
      }
    />
     
    <a
      className="btn-floating btn-small halfway-fab waves-effect waves-light blue lighten-5"
    >
      <i
        className="large material-icons indigo-text darken-4"
      >
        mode_edit
      </i>
    </a>
  </div>
  <div
    className="menu"
  >
    <Link
      className="menuItem black-text sidenav-close"
      to="/dashboard"
    >
      <h6>
        Dashboard
      </h6>
      <i
        className="material-icons"
      >
        pie_chart
      </i>
    </Link>
    <Link
      className="menuItem black-text sidenav-close"
      to="/trips/oneway"
    >
      <h6>
        Make trip request
      </h6>
      <i
        className="material-icons"
      >
        airplanemode_active
      </i>
    </Link>
    <Link
      className="menuItem black-text sidenav-close"
      to="/viewtrips/accepted"
    >
      <h6>
        My Trips
      </h6>
      <i
        className="material-icons"
      >
        card_travel
      </i>
    </Link>
  </div>
</div>
`;
